astroquery/mast/utils.py

```diff
@@ -0,0 +1,176 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+MAST Utils
+==========
+
+Miscellaneous functions used throughout the MAST module.
+"""
+
+import numpy as np
+
+import requests
+import json
+from urllib import parse
+import astropy.coordinates as coord
+
+from ..version import version
+from ..exceptions import ResolverError, InvalidQueryError
+from ..utils import commons
+
+from . import conf
+
+
+__all__ = ['resolve_object']
+
+
+def parse_type(dbtype):
+    """
+    Takes a data type as returned by a database call and regularizes it into a
+    triplet of the form (human readable datatype, python datatype, default value).
+
+    Parameters
+    ----------
+    dbtype : str
+        A data type, as returned by a database call (ex. 'char').
+
+    Returns
+    -------
+    response : tuple
+        Regularized type tuple of the form (human readable datatype, python datatype, default value).
+
+    For example:
+
+    _parse_type("short")
+    ('integer', np.int64, -999)
+    """
+
+    dbtype = dbtype.lower()
+
+    return {
+        'char': ('string', str, ""),
+        'string': ('string', str, ""),
+        'datetime': ('string', str, ""),  # TODO: handle datetimes correctly
+        'date': ('string', str, ""),  # TODO: handle datetimes correctly
+
+        'double': ('float', np.float64, np.nan),
+        'float': ('float', np.float64, np.nan),
+        'decimal': ('float', np.float64, np.nan),
+
+        'int': ('integer', np.int64, -999),
+        'short': ('integer', np.int64, -999),
+        'long': ('integer', np.int64, -999),
+        'number': ('integer', np.int64, -999),
+
+        'boolean': ('boolean', bool, None),
+        'binary': ('boolean', bool, None),
+
+        'unsignedbyte': ('byte', np.ubyte, -999)
+    }.get(dbtype, (dbtype, dbtype, dbtype))
+
+
+def _simple_request(url, params):
+    """
+    Light wrapper on requests.session().get basically to make monkey patched testing easier/more effective.
+    """
+
+    session = requests.session()
+    headers = {"User-Agent": "astroquery/{} {}".format(version, session.headers['User-Agent']),
+               "Content-type": "application/x-www-form-urlencoded",
+               "Accept": "text/plain"}
+
+    response = requests.get(url, params=params, headers=headers)
+    response.raise_for_status()
+
+    return response
+
+
+def resolve_object(objectname):
```

Comment:
This is uses the internal MAST resolver which does call Sesame, but also a number of other services so that it can resolve names that aren't resolvable through Sesame. This is also how we ensure that users get the same results they do using MAST web interfaces.

Reply:
OK to keeping it then. But consider changing the name to be more mast specific if we expose it to the user API, after all other modules do this but not with standalone functions but class methods (e.g. Simbad.query_object), etc.

Reply:
It's already exposed as part of the specific classes, which is not new behavior. I can not expose it from utils, which solve the issue of what should/should not be public in utils. The original idea was to try to move from having the resolver within the classes to just having it be a mast.utils function, but if that makes it too general, I can just scrap that idea. Especially since I need to keep it in the classes at least for the time being for backward compatibility.

Reply:
when part of the class I think it's perfectly fine. Having mast.utils is also great as it's a good way to serve multiple of the mast classes. My only issue is with this as a standalone function exposed to the API with a too generic name for a functionality that is available through many providers.
(I don't think we're going into that direction, but one can imagine to have a `resolve_object()` for astroquery with the ability to choose the backend service, whether it's e.g. mast or simbad or whatever)

Maybe @keflavich has some thoughts about it, too.

Reply:
OK, then I will keep utils entirely for within mosdule use, and have the resolver function as part of the classes as the user facing one.